grub/grub_unit_test.py

```diff
@@ -0,0 +1,93 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2024 IBM
+# Author: Krishan Gopal Saraswat <krishang@linux.ibm.com>
+
+import os
+import shutil
+from avocado import Test
+from avocado.utils import process
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class UnitTestCases(Test):
+    """
+    GRUB Unit test cases
+    """
+    def setUp(self):
+        """
+        Build Qemu and install required packages
+        Source:
+        https://github.com/qemu/qemu.git
+        """
+        sm = SoftwareManager()
+        packages = ['meson', 'gcc', 'gettext-devel', 'libisoburn', 'xorriso']
+        for package in packages:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.cancel(
+                    "Fail to install %s required for this test." % package)
+
+        # Setup Qemu
+        qemu_version = process.run("qemu-system-ppc --version",
+                                   ignore_status=True, sudo=True, shell=True)
+        if qemu_version.exit_status != 0:
+            qemu_url = "https://github.com/qemu/qemu.git"
+            process.run("git clone %s" % qemu_url)
+            # build qemu
+            cwd = os.getcwd()
+            self.sourcedir_qemu = os.path.join(cwd, "qemu")
```

Comment:
What about just  

self.sourcedir_qemu = os.path.join(os.getcwd(), "qemu") it will help to eliminate extra variable  and other places as well